Model/Order/HydrateOrderFromQuote.php

```diff
@@ -0,0 +1,198 @@
+<?php
+declare(strict_types=1);
+
+namespace Bold\Checkout\Model\Order;
+
+use Bold\Checkout\Api\Data\Http\Client\ResultInterface;
+use Bold\Checkout\Api\Http\ClientInterface;
+use Bold\Checkout\Api\Order\HydrateOrderFromQuoteInterface;
+use Bold\Checkout\Model\Order\Address\Converter;
+use Bold\Checkout\Model\Quote\GetCartLineItems;
+use Magento\Quote\Api\Data\CartInterface;
+use Magento\Quote\Model\Quote\Address\ToOrderAddress;
+
+/**
+ * Hydrate Bold order from Magento quote.
+ */
+class HydrateOrderFromQuote implements HydrateOrderFromQuoteInterface
+{
+    private const HYDRATE_ORDER_URL = 'checkout_sidekick/{{shopId}}/order/%s';
+    private const EXPECTED_SEGMENTS = [
+        'subtotal',
+        'shipping',
+        'discount',
+        'tax',
+        'grand_total',
+    ];
+
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var GetCartLineItems
+     */
+    private $getCartLineItems;
+
+    /**
+     * @var Converter
+     */
+    private $addressConverter;
+
+    /**
+     * @var ToOrderAddress
+     */
+    private $quoteToOrderAddressConverter;
+
+    /**
+     * @param ClientInterface $client
+     * @param GetCartLineItems $getCartLineItems
+     * @param Converter $addressConverter
+     * @param ToOrderAddress $quoteToOrderAddressConverter
+     */
+    public function __construct(
+        ClientInterface $client,
+        GetCartLineItems $getCartLineItems,
+        Converter $addressConverter,
+        ToOrderAddress $quoteToOrderAddressConverter,
+    ) {
+        $this->client = $client;
+        $this->getCartLineItems = $getCartLineItems;
+        $this->addressConverter = $addressConverter;
+        $this->quoteToOrderAddressConverter = $quoteToOrderAddressConverter;
+    }
+
+    /**
+     * @param CartInterface $quote
+     * @param string $publicOrderId
+     * @return ResultInterface
+     */
+    public function hydrate(CartInterface $quote, string $publicOrderId): ResultInterface
+    {
+        $websiteId = (int)$quote->getStore()->getWebsiteId();
+        $billingAddress = $this->quoteToOrderAddressConverter->convert($quote->getBillingAddress());
+
+        if ($quote->getIsVirtual()) {
+            $totals = $quote->getBillingAddress()->getTotals();
```

Comment:
👏🏽 Ah, I forgot this piece. 
[Clarification] So Magento doesn't handle the difference between this? Trying to understand why the totals would be different?